### PR TITLE
Feature/DEV-491 encrypt attachments

### PR DESCRIPTION
**Pull request by Waseem826**

### CODE REVIEW

File: src/app/mail/mail-detail/mail-detail.component.html [hunk 1]
```diff
@@ -325,15 +325,28 @@ <h5 class="ui-header-subtitle text-dark mb-0">
...
+             [href]="attachment.document" placement="top" [ngbTooltip]="attachment.document | filename" target="_blank"
+             rel="noopener">
             <div class="attachment-icon">
               <i class="icon icon-attachment"></i>
             </div>
             <div class="attachment-name">
               {{attachment.document | filename}}
             </div>
           </a>
+          <ng-template #encryptedAttachment>
```
Comment (atifsaddique211f) [inline src/app/mail/mail-detail/mail-detail.component.html]: we have two templates by same name, maybe we can keep one on this page and use on both places.

Comment (atifsaddique211f) [reply src/app/mail/mail-detail/mail-detail.component.html]: And we can pass the email (`mail` or `childMail`) as an argument to template

File: src/app/store/reducers/compose-mail.reducers.ts [hunk 2]
```diff
@@ -204,6 +239,8 @@ export function reducer(state: ComposeMailState = { drafts: {} }, action: Compos
           }
           return true;
         });
+      const isProcessingAttachments = !!state.drafts[action.payload.draftId].attachments.find(attachment => attachment.inProgress);
```
Comment (atifsaddique211f) [inline src/app/store/reducers/compose-mail.reducers.ts]: this chunk of code is repeated on many places, maybe we should put it in a private method.

File: src/app/mail/mail-detail/mail-detail.component.html (hunk 1 above)
Comment (Waseem826) [reply src/app/mail/mail-detail/mail-detail.component.html]: Yes you're right

**Pull request merged**